### PR TITLE
Fix broken relay-experimental imports; Add Record: RelayModernRecord to relay-runtime API

### PR DESCRIPTION
**Pull request by babangsund**
Suggested fix for https://github.com/facebook/relay/issues/2845

### CODE REVIEW

File: packages/relay-experimental/useRelayEnvironment.js [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 'use strict';
 
-const ReactRelayContext = require('react-relay/ReactRelayContext');
+const {ReactRelayContext} = require('react-relay');
```
Comment (jstejada) [inline packages/relay-experimental/useRelayEnvironment.js]: the problem here is that we don't actually want to bring in all of 'react-relay' in our web bundle, let me see if I can get this to work

Comment (babangsund) [reply packages/relay-experimental/useRelayEnvironment.js]: Hadn't considered this aspect. 🤔

Comment (jstejada) [reply packages/relay-experimental/useRelayEnvironment.js]: I think we still want to be compatible with the apis in `react-relay`, so you can use both apis together and migrate incrementally to hooks if you want (without having to convert everything to hooks at once), which means that we need to use the same context; but for cases where we're exclusively using relay-experimental we don't to bring in all of react-relay.

I think I might have a solution for this though

Comment (babangsund) [reply packages/relay-experimental/useRelayEnvironment.js]: Okay. Looking forward to being blown away. 😉

**Pull request closed without merging**